src/aiida/orm/nodes/data/data.py

```diff
@@ -44,6 +45,10 @@ class Data(Node):
     _storable = True
     _unstorable_message = 'storing for this node has been disabled'
 
+    __qb_fields__: Sequence[QbField] = (
```

Comment:
This is an interesting case.

- `Node` defines `__qb_fields__` as `Sequence[QbField]`.
- `Data(Node)` proceeds to assign `(QbAttrField(...), ...)` to it, which is fine, as `QbAttrField` is a child of `QbField`
- `Dict(Data)` assigns `(QbField(...), ...)` to it - `mypy` complains, as `QbField` is not a child of `QbAttrField`

I circumvented the issue by explicitly redefining the type in `Data`, as well as in `RemoteFolderData`. Again, I suspect there is a better approach, perhaps using generics, etc.

Reply:
Changed to lists throughout